include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp

```diff
@@ -272,4 +285,99 @@ inline typename seq_type::iterator erase_gap(seq_type & seq, typename seq_type::
+ * \tparam alignment_t  The type of the alignment, must satisfy tuple_like_concept.
+ * \tparam idx          An index sequence.
+ * \param[in] stream    The output stream that receives the formatted alignment.
+ * \param[in] align     The alignment that shall be streamed.
+ */
+template<tuple_like_concept alignment_t, size_t ...idx>
+void stream_alignment(debug_stream_type & stream, alignment_t const & align, std::index_sequence<idx...> const & /**/)
+{
+    size_t const alignment_length = std::get<0>(align).size();
+    char const * indent = "        ";
+
+    // split alignment into blocks of length 50 and loop over parts
+    for (size_t used_length = 0; used_length < alignment_length; used_length += 50)
+    {
+        // write header
+        if (used_length != 0)
+            stream << std::endl;
+
+        stream << std::setw(7) << used_length << ' ';
+        for (size_t col = 1; col <= 50 && col + used_length <= alignment_length; ++col)
+        {
+            if (col % 10 == 0)
+                stream << ':';
+            else if (col % 5 == 0)
+                stream << '.';
+            else
+                stream << ' ';
+        }
+
+        // write sequences
+        stream << std::endl << indent;
+        size_t const col_end = std::min(used_length + 50, alignment_length);
+        ranges::for_each(std::get<0>(align) | ranges::view::slice(used_length, col_end) | view::to_char,
+                         [&stream] (char ch) { stream << ch; });
+
+        auto stream_f = [&] (auto const & previous_sequence, auto const & aligned_sequence)
+        {
+            stream << std::endl << indent;
+            auto seq1 = previous_sequence.begin() + used_length;
+            auto seq2 = aligned_sequence.begin() + used_length;
+            for (auto it1 = seq1, it2 = seq2;
+                 it1 < previous_sequence.end() && it1 < seq1 + 50 &&
+                 it2 < aligned_sequence.end()  && it2 < seq2 + 50;
+                 ++it1, ++it2)
+            {
+                stream << (seqan3::to_char(*it1) == seqan3::to_char(*it2) ? '|' : ' ');
+            }
+            stream << std::endl << indent;
+            ranges::for_each(aligned_sequence | ranges::view::slice(used_length, col_end) | view::to_char,
+                             [&stream] (char ch) { stream << ch; });
+        };
+        (stream_f(std::get<idx>(align), std::get<idx + 1>(align)), ...);
+        stream << std::endl;
+    }
+}
+
+} // namespace detail
+
+/*!
+ * \brief True, if each type satisfies aligned_sequence_concept; false otherwise.
+ * \tparam elems The pack of types to be tested.
+ */
+template <typename ...elems>
+inline bool constexpr all_satisfy_aligned_seq = false;
+
+/*!
+ * \brief True, if each type satisfies aligned_sequence_concept; false otherwise.
+ * \tparam elems The pack of types to be tested.
+ */
+template <typename ...elems>
+inline bool constexpr all_satisfy_aligned_seq<type_list<elems...>> = (aligned_sequence_concept<elems> && ... && true);
```

Comment:
isn't
```suggestion
inline bool constexpr all_satisfy_aligned_seq = (aligned_sequence_concept<elems> && ... );
```
enough to make it work?